src/repository.c

```diff
@@ -2061,6 +2058,43 @@ static int repo_init_directories(
 	return error;
 }
 
+static int repo_init_head(const char *repo_dir, const char *given)
+{
+	git_config *cfg = NULL;
+	git_buf head_path = GIT_BUF_INIT, cfg_branch = GIT_BUF_INIT;
+	const char *initial_head = NULL;
+	int error;
+
+	if ((error = git_buf_joinpath(&head_path, repo_dir, GIT_HEAD_FILE)) < 0)
+		goto out;
+
+	/*
+	 * A template may have set a HEAD; use that unless it's been
+	 * overridden by the caller's given initial head setting.
+	 */
+	if (git_path_exists(head_path.ptr) && !given)
```

Comment:
This kind of crosses my current effort to stop using direct file access whenever writing symbolic references. We really shouldn't be doing that at all and always use the refdb. I realize we're not quite there yet when initializing new repositories as there may not yet be a refdb, but we could maybe do this as a preliminary refactoring.

Reply:
I just moved this, it's not net new.  I would expect the conflict would be reasonably easy to resolve?  It looks like #5563 is ready to merge so, in fact, it seems like _my_ conflict to resolve?